torch/_dynamo/variables/builder.py

```diff
@@ -715,6 +721,12 @@ def index_source(key):
                 source=self.source,
                 guards=make_guards(GuardBuilder.ID_MATCH),
             )
+        elif issubclass(type(value), type) and is_torch_ctx_manager_class(value):
```

Comment:
What's the type subclass test for?

Reply:
Oh, it's not necessary any more, will remove it.

Reply:
I think we still need it, because we are checking if value is in the dict, the value should be hashable.